### PR TITLE
Description Enum Refactoring

### PR DESCRIPTION
**Pull request by jGleitz**
:warning: Do not merge this PR before #123 is merged!

As discussed in #123:

> I have divided the `Description` enums into types where they are used. This should either be placed into `DescriptionPathAssertion` or into `DescriptionBasic` if it is something we want to use in other places as well.
Maybe it makes even sense to move `TO_BE` and `NOT_TO` to `DescriptionBasic` - up to you if you want to do that as well

I moved all four constants to `DescriptionBasic`. `TO_BE` and `NOT_TO_BE` still exist in `DescriptionAnyAssertion` for backwards compatibility.

----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by robstoll** [comment]
Perfect, would you like to do the same for de_CH Descriptions in translations/de_CH/?

**Comment by jGleitz** [comment]
> Perfect, would you like to do the same for de_CH Descriptions in translations/de_CH/? 

I am not sure what you mean, I made the change in `DescriptionAnyAssertion` in both `en_GB` and `de_CH`.

**Comment by robstoll** [comment]
> I am not sure what you mean, I made the change in DescriptionAnyAssertion in both en_GB and de_CH.

Sorry, my mistake, I saw only en_GB this morning

File: core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-ch.tutteli.atrium.translations.DescriptionAnyAssertion-TO_BE = est
-ch.tutteli.atrium.translations.DescriptionAnyAssertion-NOT_TO_BE = n'est pas
-ch.tutteli.atrium.translations.DescriptionAnyAssertion-IS_NOT_SAME = n'est pas la m�me instance que
+ch.tutteli.atrium.translations.DescriptionBasic-TO_BE = est
+ch.tutteli.atrium.translations.DescriptionBasic-NOT_TO_BE = n'est pas
+ch.tutteli.atrium.translations.DescriptionAnyAssertion-IS_NOT_SAME = n'est pas la m\u00EAme instance que
 ch.tutteli.atrium.specs.AssertionVerb-ASSERT = il applique que
-ch.tutteli.atrium.specs.reporting.translating.TranslationSupplierSpec$TestTranslatable-DATE_KNOWN=%tD �tait %<tA!!
-ch.tutteli.atrium.specs.reporting.translating.TranslationSupplierSpec$TestTranslatable-PLACEHOLDER=Caract�re de remplacement %s
+ch.tutteli.atrium.specs.reporting.translating.TranslationSupplierSpec$TestTranslatable-DATE_KNOWN=%tD �tait %<tA!!
```
Comment (robstoll) [inline core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties]: That does not look right, but specs are green, so I guess its OK :laughing:

Comment (jGleitz) [reply core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties]: Does not look right at all!
I had encoding issues with même, which I could only solve by replacing it with an escaped unicode character. I am curios why there is no test failure for this one…

Comment (robstoll) [reply core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties]: The problem is actually intellij, it saves properties file as UTF-8 where resource-bundle properties should be saved as ISO-8859-1

Comment (robstoll) [reply core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties]: I found the problem, spek1 has the problem that it does not execute two branches with the same name. Could you please fix this and the next line and I will point out in another comment where the fix needs to apply

Comment (jGleitz) [reply core/robstoll-lib/atrium-core-robstoll-lib-jvm/src/test/resources/ch/tutteli/atrium/core/robstoll/lib/reporting/translating/Atrium_fr.properties]: Done. But please see comment below.

**Comment by jGleitz** [comment]
Note: `Atrium_it.properties` and `TranslatorIntSpec$TestTranslatable_it.properties` are still not used anywhere (changing them does not make any test fail). Is this another case of a test case not being executed?

**Comment by robstoll** [comment]
Thanks for the PR, I'll look into the issue with TranslatorIntSpec$TestTranslatable_it.properties and look for a fix for PropertiesPerEntityAndLocaleTranslationSupplier. 
As I plan to move away from ResourceBundle latest with 1.0.0 anyway (I'll probably switch to gettext) I am not even sure if it is worth to fix it.

**Pull request merged**